lib/bigtable/filter.js

```diff
@@ -0,0 +1,896 @@
+ *     }
+ *   }
+ * ];
+ *
+ * //-
+ * // <h4>Row Samples</h4>
+ * //
+ * // Matches all cells from a row with probability p, and matches no cells
+ * // from the row with probability 1-p.
+ * //-
+ * var filter = [
+ *   {
+ *     row: {
+ *       sample: 1
+ *     }
+ *   }
+ * ];
+ *
+ * //-
+ * // <h4>Row Cell Offsets</h4>
+ * //
+ * // Skips the first N cells of each row, matching all subsequent cells.
+ * // If duplicate cells are present, as is possible when using an
+ * // {module:bigtable/filter#interleave}, each copy of the cell is counted
+ * // separately.
+ * //-
+ * var filter = [
+ *   {
+ *     row: {
+ *       cellOffset: 2
+ *     }
+ *   }
+ * ];
+ *
+ * //-
+ * // <h4>Row Cell Limits/h4>
+ * //
+ * // Matches only the first N cells of each row.
+ * // If duplicate cells are present, as is possible when using an
+ * // {module:bigtable/filter#interleave}, each copy of the cell is counted
+ * // separately.
+ * //-
+ * var filter = [
+ *   {
+ *     row: {
+ *       cellLimit: 4
+ *     }
+ *   }
+ * ];
+ */
+Filter.prototype.row = function(row) {
+  if (!is.object(row)) {
+    row = {
+      key: row
+    };
+  }
+
+  if (row.key) {
+    var key = Filter.convertToRegExpString(row.key);
+
+    key = Mutation.convertToBytes(key);
+    this.set('rowKeyRegexFilter', key);
+  }
+
+  if (row.sample) {
+    this.set('rowSampleFilter', row.sample);
+  }
+
+  if (is.number(row.cellOffset)) {
+    this.set('cellsPerRowOffsetFilter', row.cellOffset);
+  }
+
+  if (is.number(row.cellLimit)) {
+    this.set('cellsPerRowLimitFilter', row.cellLimit);
+  }
+
+  return this;
+};
+
+/**
+ * Stores a filter object.
+ *
+ * @private
+ *
+ * @param {string} key - Filter name.
+ * @param {*} value - Filter value.
+ */
+Filter.prototype.set = function(key, value) {
+  var filter = {};
+
+  filter[key] = value;
+  this.filters_.push(filter);
+  return this;
+};
+
+/**
+ * This filter is meant for advanced use only. Hook for introspection into the
+ * filter. Outputs all cells directly to the output of the read rather than to
+ * any parent filter.
+ *
+ * Despite being excluded by the qualifier filter, a copy of every cell that
+ * reaches the sink is present in the final result.
+ *
+ * As with an {module:bigtable/filter#interleave} filter, duplicate cells are
+ * possible, and appear in an unspecified mutual order.
+ *
+ * Cannot be used within {module:bigtable/filter#condition} filter.
+ *
+ * @example
+ * //-
+ * // Using the following filter, a copy of every cell that reaches the sink is
+ * // present in the final result, despite being excluded by the qualifier
+ * // filter
+ * //-
+ * var filter = [
+ *   {
+ *     family: 'follows'
+ *   }, {
+ *     interleave: [
+ *       [
+ *         {
+ *           all: true
+ *         }
+ *       ], [
+ *         {
+ *           label: 'prezzy'
+ *         }, {
+ *           sink: true
+ *         }
+ *       ]
+ *     ]
+ *   }, {
+ *     column: 'gwashington'
+ *   }
+ * ];
```

Comment:
I think it might be more readable even more expanded:

```js
var filter = [
  {
    family: 'follows'
  },
  {
    interleave: [
      [
        {
          all: true
        }
      ],
      [
        {
          label: 'prezzy'
        },
        {
          sink: true
        }
      ]
    ]
  },
  {
    column: 'gwashington'
  }
];
```